File: src/JavaScript/easy/dynamic programming/problem338.js
/**
 * @param {number} n
 * @return {number[]}
 */

var countBits = function(n) {
    let ans = []
    for(let i = 0; i <= n; i++){
        ans.push((i.toString(2).match(/1/g) || []).length);
    }
   return ans
};

// let n = 2
let n = 5

countBits(n)

File: src/Python/easy/dynamic programming/problem338.py
from typing import List

class Solution:
    def countBits(self, n: int) -> List[int]:
        ans = []
        for i in range(n + 1):
            ans.append(bin(i).replace("0b", "").count("1"))
        return ans


n = 5

solution = Solution()
solution.countBits(n)